code/controllers/Processes/spacedrift.dm

```diff
@@ -0,0 +1,57 @@
+#define MAX_THROWING_DIST 512 // 2 z-levels on default width
+#define MAX_TICKS_TO_MAKE_UP 3 //how many missed ticks will we attempt to make up for this run.
+
+var/global/datum/controller/process/spacedrift/drift_master
+
+/datum/controller/process/spacedrift
+	var/list/processing_list = list()
+
+/datum/controller/process/spacedrift/setup()
+	name = "spacedrift"
+	schedule_interval = 5
+	start_delay = 20
+	log_startup_progress("Spacedrift starting up.")
+	if(drift_master)
+		qdel(drift_master)
+	drift_master = src
+
+/datum/controller/process/spacedrift/statProcess()
+	..()
+	stat(null, "P:[processing_list.len]")
+
+/datum/controller/process/spacedrift/doWork()
+	var/list/currentrun = processing_list.Copy()
+
+	while(currentrun.len)
```

Comment:
I'd rather not touch these any further than I already have, they're easy to make nonfunctional